### PR TITLE
Use GHA paths

### PR DESCRIPTION
**Pull request by paulb777**
GitHub Actions has built in path support to trigger workflows if certain files have changed. See https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#onpushpull_requestpaths

Starting to use this instead of the if_changed.sh script we use with travis.

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  check:
+    runs-on: macOS-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Check
+      run: |
+        brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/e3496d9/Formula/clang-format.rb
+        brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/7963c3d/Formula/swiftformat.rb
+        pip install flake8
+        ./scripts/check.sh --test-only
```
Comment (paulb777) [inline .github/workflows/check.yml]: Still need to verify this fully works with GHA

File: .github/workflows/core.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: core
+
+on:
+  pull_request:
+    paths:
+    - 'FirebaseCore**'
```
Comment (ryanwilson) [inline .github/workflows/core.yml]: Do we also want to test when GoogleUtilities changes?

File: .github/workflows/dynamiclinks.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: dynamiclinks
+
+on:
+  pull_request:
+    paths:
+    - 'Firebase/DynamicLinks**'
```
Comment (ryanwilson) [inline .github/workflows/dynamiclinks.yml]: Should we also test on Core changes?

**Comment by paulb777** [comment]
My preference is defer on triggering on all dependencies until we have more complete tests in place and are better able to assess the time/benefit trade-offs.

**Pull request merged**